impeller/typographer/backends/skia/text_render_context_skia.cc

```diff
@@ -340,25 +340,39 @@ static std::shared_ptr<Texture> UploadGlyphTextureAtlas(
 
 std::shared_ptr<GlyphAtlas> TextRenderContextSkia::CreateGlyphAtlas(
     GlyphAtlas::Type type,
+    std::shared_ptr<GlyphAtlasContext> atlas_context,
     FrameIterator frame_iterator) const {
   TRACE_EVENT0("impeller", __FUNCTION__);
   if (!IsValid()) {
     return nullptr;
   }
-
-  auto glyph_atlas = std::make_shared<GlyphAtlas>(type);
+  auto last_atlas = atlas_context->GetGlyphAtlas();
 
   // ---------------------------------------------------------------------------
   // Step 1: Collect unique font-glyph pairs in the frame.
   // ---------------------------------------------------------------------------
 
   auto font_glyph_pairs = CollectUniqueFontGlyphPairs(type, frame_iterator);
   if (font_glyph_pairs.empty()) {
-    return glyph_atlas;
+    return last_atlas;
   }
 
   // ---------------------------------------------------------------------------
-  // Step 2: Get the optimum size of the texture atlas.
+  // Step 2: Determine if the atlas type and font glyph pairs are identical to
+  //         the current atlas and reuse if possible.
+  // ---------------------------------------------------------------------------
+  if (last_atlas->GetType() == type) {
+    auto new_glyphs = last_atlas->FindNewGlyphs(font_glyph_pairs);
```

Comment:
I'm not sure if this is actually good enough of a check. If glyphs are drawn at different sizes I think we'd need to account for that ... right? But based on my reading of the code its only one FontGlyphPair mapped to each rect, so a different size glyph has to be a unique font glyph pair ... right?

Reply:
If the glyph size changes the rect won't match.